spinedb_api/db_mapping_base.py

```diff
@@ -720,13 +813,84 @@ def key(self):
             return None
         return (self._item_type, id_)
 
-    @property
-    def is_id_valid(self):
-        return self._is_id_valid
+    def same_db_item(self, db_item):
+        """Tests if database item that has same db id is in fact same as this item.
+
+        Args:
+            db_item (dict): item fetched from database
+
+        Returns:
+            bool: True if items are the same, False otherwise
+        """
+        raise NotImplementedError()
```

Comment:
I feel `same_db_item` could have a generic implementation based on some attributes the subclass would need to define. Is the subclass implementation really different across different subclasses? If all do more or less the same I would consider doing the work right here. That might be easier to maintain and also more consistent with the rest of the class (we use that strategy for many other methods too).